pkgs/by-name/md/mdfried/package.nix

```
@@ -0,0 +1,38 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  testers,
+}:
+
+rustPlatform.buildRustPackage (finalAttrs: rec {
+  pname = "mdfried";
+  version = "0.12.1";
+
+  src = fetchFromGitHub {
+    owner = "benjajaja";
+    repo = "mdfried";
+    rev = "v${version}";
+    hash = "sha256-pSJexHOfGB8KGTpPrqw+dgymDXyux0uH6CDsZcnsHlE=";
+  };
+
+  cargoHash = "sha256-ZcWoYfvYmesi7JPOeSmIj0L9qlsoOYf6SMO0XQy6KwA=";
+
+  doCheck = true;
+
+  passthru.tests = {
+    version = testers.testVersion {
+      package = finalAttrs.finalPackage;
+      command = "mdfried --version";
+    };
+  };
```

```suggestion
  doInstallCheck = true;
  nativeInstallCheckInputs = [ versionCheckHook ];
```

This blocks the build and doesn't bring in a separated derivation (that sometimes can be omitted), and is preferred now